### PR TITLE
[LW-9597] Transaction expiry date

### PR DESCRIPTION
**Pull request by lucas-barros**
# Checklist

- [x] JIRA - https://input-output.atlassian.net/browse/LW-9597
- [x] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Add expiry date for send flow and dapp transaction confirmation.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

image

image

### CODE REVIEW

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx [hunk 1]
```diff
@@ -162,6 +164,23 @@ export const DappTransaction = ({
           />
         )}
 
+        <div className={styles.depositContainer}>
```
Comment (VanessaPC) [inline packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:167]: Is it possible to move this to it's own component in package UI, most of the data in this component gets passed down to the UI and it's more clear.

File: packages/core/src/ui/components/OutputSummaryList/OutputSummaryList.tsx [hunk 2]
```diff
@@ -65,6 +76,25 @@ export const OutputSummaryList = ({
       )}
 
       <div className={styles.feeContainer} data-testid="summary-fee-container">
+        <RowContainer>
```
Comment (VanessaPC) [inline packages/core/src/ui/components/OutputSummaryList/OutputSummaryList.tsx:79]: Then you can reuse here.

File: packages/ui/src/design-system/dapp-transaction-summary/dapp-transaction-field.component.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import type { ReactNode } from 'react';
```
Comment (VanessaPC) [inline packages/ui/src/design-system/dapp-transaction-summary/dapp-transaction-field.component.tsx:1]: 'transactionField' could be any field in the transaction, can we be more specific in this one?

**Review by VanessaPC — CHANGES_REQUESTED** [review]
A few small comments, also, what does it happen when a transaction validity interval is expired?

File: packages/core/src/ui/components/OutputSummaryList/OutputSummaryList.tsx [hunk 4]
```diff
@@ -65,6 +76,25 @@ export const OutputSummaryList = ({
...
+            label: translations.expiresBy,
+            tooltipContent: translations.expiresByTooltip,
+            dataTestId: 'validity-interval-expires-by-label'
+          })}
+
+          <Text className={styles.validityIntervalExpiresBy} data-testid="validity-interval-expires-by-value">
+            {expiresBy ? (
+              <>
+                <span>{expiresBy.utcDate}</span>
+                <span>{expiresBy.utcTime} UTC</span>
```
Comment (VanessaPC) [inline packages/core/src/ui/components/OutputSummaryList/OutputSummaryList.tsx:90]: UTC is string, let's move it to translations :)

Comment (lucas-barros) [reply packages/core/src/ui/components/OutputSummaryList/OutputSummaryList.tsx]: Fixed here: https://github.com/input-output-hk/lace/pull/1008/commits/aeba5afe56fd05b746a775dc8e3d22b240edeb95

File: packages/ui/src/design-system/dapp-transaction-summary/dapp-transaction-field.component.tsx (hunk 3 above)
Comment (lucas-barros) [reply packages/ui/src/design-system/dapp-transaction-summary/dapp-transaction-field.component.tsx]: Fixed here: https://github.com/input-output-hk/lace/pull/1008/commits/3e1a2cb1512e6c6e8e5b8fb70f7ec92be1b09a01

File: packages/core/src/ui/components/OutputSummaryList/OutputSummaryList.tsx (hunk 2 above)
Comment (lucas-barros) [reply packages/core/src/ui/components/OutputSummaryList/OutputSummaryList.tsx:80]: There are some styling diferences between send flow confirmation and dapp confirmation. In my opnion, to use the UI components here we need to remove all the local components here and replace with TransactionSummary components.

**Comment by lucas-barros** [comment]
> A few small comments, also, what does it happen when a transaction validity interval is expired? currently the dapp-connector doesn't open.

I'm not sure. The submitted transaction should fail, but I don't know why dapp connector doesnt open.

**Comment by VanessaPC** [comment]
> > A few small comments, also, what does it happen when a transaction validity interval is expired? currently the dapp-connector doesn't open.
> 
> I'm not sure. The submitted transaction should fail, but I don't know why dapp connector doesnt open.

It does fail.. but silently. That's on your branch. If you want to see it yourself, I can tell you where the dapp connector is.. 
Screenshot 2024-04-03 at 16 24 55

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx (hunk 1 above)
Comment (VanessaPC) [reply packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:167]: I saw it now, then I'd say either extract this out so it's clear and it reads nice, 
```
const  getExpireByText = expiresBy ? (
                <Flex flexDirection="column" alignItems="flex-end">
                  <span>{expiresBy.utcDate}</span>
                  <span>{expiresBy.utcTime} UTC</span>
                </Flex>
              ) : (
                t('core.outputSummaryList.noLimit')
              )
     ```

File: apps/browser-extension-wallet/src/utils/__tests__/summary-slot-datetime.test.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    parameters: { epochLength: 86_400, slotLength: 1000 },
+    start: { slot: 0, time: { __type: 'Date', value: 1_666_656_000_000 } }
+  }
+]);
+
+const testSlot = Cardano.Slot(36_201_583);
+
+describe('Testing summarySlotDatetime', () => {
+  test('should return undefined', async () => {
+    expect(summarySlotDatetime(undefined, testSlot)).toEqual(undefined);
```
Comment (DominikGuzei) [inline apps/browser-extension-wallet/src/utils/__tests__/summary-slot-datetime.test.ts:21]: Let's add the second case where slot is undefined here

File: apps/browser-extension-wallet/src/utils/summary-slot-datetime.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable consistent-return */
+import { EraSummary } from '@cardano-sdk/core';
+import { Wallet } from '@lace/cardano';
+import { formatDate, formatTime } from './format-date';
+
+export const summarySlotDatetime = (
```
Comment (DominikGuzei) [inline apps/browser-extension-wallet/src/utils/summary-slot-datetime.ts:6]:
```suggestion
export const eraSlotDateTime
```

**Review by DominikGuzei — APPROVED** [review]

File: apps/browser-extension-wallet/src/utils/summary-slot-datetime.ts (hunk 6 above)
Comment (lucas-barros) [reply apps/browser-extension-wallet/src/utils/summary-slot-datetime.ts]: https://github.com/input-output-hk/lace/pull/1008/commits/4a57cf1788f4cb683cafbbcbfef3c8be255840b9

File: apps/browser-extension-wallet/src/utils/__tests__/summary-slot-datetime.test.ts (hunk 5 above)
Comment (lucas-barros) [reply apps/browser-extension-wallet/src/utils/__tests__/summary-slot-datetime.test.ts]: https://github.com/input-output-hk/lace/pull/1008/commits/0b842caa2a2d5183eceabb06fe198f11a932c4a3

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx (hunk 1 above)
Comment (lucas-barros) [reply packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:178]: https://github.com/input-output-hk/lace/pull/1008/commits/9b5efa1135dbd9d932acba3605a23bcceb3c99ff

**Review by VanessaPC — APPROVED** [review]
Nice 🚀 !

**Review by ljagiela — APPROVED** [review]

**Pull request merged**